### PR TITLE
feat: diag tool supports sub commands and hosts & refactor

### LINKED ISSUE #2929: diag: 0.7.0 use sbin to start cluster, diag should be able to parse conf/host

### LINKED ISSUE #2972: diag: support sub commands
We can use diag tool by connect zk. Only pull config and logs needs dist yaml/hosts.

e.g.  we can use 
```
openmldb_tool inspect offline --cluster=x.x.x.x:2181/openmldb
```
to check all offline jobs.

`--cluster` default is 127.0.0.1:2181/openmldb, so in demo docker, we can run it easily:
```
openmldb_tool status
openmldb_tool inspect online
```

And about `inspect online`, we can use `show table status`, but it'll skip hidden db.

### PR DESCRIPTION
**Pull request by vagetablechicken**
close #2929, close #2972

### CODE REVIEW

File: docs/zh/maintain/diagnose.md [hunk 1]
```diff
@@ -2,35 +2,119 @@
...
+```
+openmldb_tool inspect -h
+usage: openmldb_tool inspect [-h] [--helpfull] {online,offline} ...
+
+positional arguments:
+  {online,offline}
+    online          only inspect online table
+    offline         only inspect offline jobs, check the job log
+```
+在线检查会检查集群中的表状态（包括系统表），并输出有异常的表，包括表的状态，分区信息，副本信息等，等价于`SHOW TABLE STATUS`并筛选出有异常的表。如果发现集群表现不正常，请先检查下是否有异常表。例如，`SHOW JOBS`无法正常输出历史任务时，可以`inspect online`检查一下是否是job系统表出现问题。
```
Comment (vagetablechicken) [inline docs/zh/maintain/diagnose.md:70]: show table status like % to show hidden db tables

File: docs/zh/maintain/diagnose.md [hunk 2]
```diff
@@ -2,35 +2,119 @@
...
+localhost:10922 /tmp/openmldb/tablet-2
+
+[nameserver]
+localhost:7527
+
+[apiserver]
+localhost:9080
+
+[taskmanager]
+localhost:9902
```
Comment (zhanghaohit) [inline docs/zh/maintain/diagnose.md:112]: the default deploy directory is `OPENMLDB_HOME`, which is defaulted to the current openmldb release directory.

```
OPENMLDB_HOME="$(cd "$(dirname "$0")"/.. || exit 1; pwd)"
```

Alternatively, we can update the deploy scripts to generate a hosts_deployed with all the detailed informations.

Comment (vagetablechicken) [reply docs/zh/maintain/diagnose.md:112]: We can set the default deploy dir by `--default_dir` in diag tool, users can set it manually now, default value is `/work/openmldb` for demo docker. `hosts_deployed` is a great idea, we can improve in the future

**Review by aceforeverd — APPROVED** [review]

File: python/openmldb_tool/tests/hosts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+[nameserver]
+localhost:7527
+
+[apiserver]
+localhost:9080
+
+[zookeeper]
+localhost:2181:2888:3888 /tmp/openmldb/zk-1
+# no taskmanager, wait https://github.com/4paradigm/OpenMLDB/issues/2933
```
Comment (zhanghaohit) [inline python/openmldb_tool/tests/hosts]: can remove these comments

**Review by zhanghaohit — APPROVED** [review]

**Pull request merged**